services/ui/ws/window_tree.cc

```diff
@@ -900,6 +900,13 @@ void WindowTree::SendToPointerWatcher(const ui::Event& event,
                                    display_id);
 }
 
+WindowManagerState* WindowTree::AppendExternalModeWindowManagerState(
```

Comment:
Ok, I was going to test/modify your patch to merge, and it crashed.

I fixed with the following:

diff --git a/services/ui/ws/display.cc b/services/ui/ws/display.cc
index 5a46408..f360a8b 100644
--- a/services/ui/ws/display.cc
+++ b/services/ui/ws/display.cc
@@ -223,10 +223,11 @@ void Display::InitDisplayRoot() {
   WindowTree* window_tree = window_server_->GetTreeForExternalWindowMode();
   ServerWindow* server_window = display_root_ptr->root();
 
-  display_root_ptr->window_manager_state_ =
+  WindowManagerDisplayRoot* display_root = display_root_ptr.get();
+  display_root->window_manager_state_ =
       window_tree->AppendExternalModeWindowManagerState(
           base::MakeUnique<WindowManagerState>(window_tree));
-  display_root_ptr->window_manager_state_->AddWindowManagerDisplayRoot(
+  display_root->window_manager_state_->AddWindowManagerDisplayRoot(
       std::move(display_root_ptr));
 
   window_tree->AddRoot(server_window);

is it ok with you, fred?